### ISSUE TITLE
HRESULT E_FAIL COM component

### ISSUE THREAD

**Issue by ramtinak** [lukasf/FFmpegInterop#48]
Hi,
We are working on Instagram client. for play broadcasting an live video, I used this peace of codes:

```
FFmpegInteropConfig FFmpegConfig;
FFmpegInteropMSS ffmpeg;
MediaStreamSource mss;

private async void play()
{
FFmpegConfig = new FFmpegInteropConfig
{
PassthroughAudioAAC = false,
PassthroughAudioMP3 = false,
PassthroughVideoH264 = true,
PassthroughVideoH264Hi10P = false,
PassthroughVideoHEVC = false,
VideoOutputAllowIyuv = false
};
var url = "rtmp://svelivestream007.16.ash1.facebook.com:16000/live-hd/17913591016173754";
ffmpeg = await FFmpegInteropMSS.CreateFromUriAsync(url, FFmpegConfig);
mss = ffmpeg.GetMediaStreamSource();
Element.SetMediaStreamSource(mss);
Element.SetPlaybackSource(MediaSource.CreateFromUri(new Uri(url)));
Element.Play();
}
```

But CreateFromUriAsync returns null with this error:
```Error HRESULT E_FAIL has been returned from a call to a COM component```

Here is my rtmp url example:
```
rtmp://svelivestream007.16.ash1.facebook.com:16000/live-hd/17913591016173754
```
what did I wrong?

**Comment by mcosmin222** [comment]
I've had this issue on some URI. FFMPEG context initialization fails at some point. You need to see where it fails and why. The error code returned can be used to determine the problem.

You can use this API to see what the error means

https://www.ffmpeg.org/doxygen/3.4/group__lavu__error.html

**Comment by ramtinak** [comment]
I just find out the rtmp url that instagram gives me it's not works at all.
I notice Instagram has two other urls, 1. DashPlaybackUrl 2. DashAbrPlaybackUrl.
I find out for play stream over Dash, you should use AdaptiveMediaSource code. which not working either! 
```            var streamUri = new Uri(t.DashPlaybackUrl);
            var streamResponse = await AdaptiveMediaSource.CreateFromUriAsync(streamUri);
            if (streamResponse.Status == AdaptiveMediaSourceCreationStatus.Success)
            {
                Player.SetMediaSource(streamResponse.MediaSource);
                Player.Play();
            }
            else
            {
                Debug.WriteLine("not found");
            }```

for FFMpeginterop still gives me `HRESULT E_FAIL has been returned from a call to a COM component` error.
one more thing, VLC(uwp version) can play these dash urls. why is that?

Should I add some parameters to FFMpegInterop to support ?

**Comment by mcosmin222** [comment]
The E_FAIL does not come from FFMPEG. This is a COM error. The actual problem happens sometime before the exception is thrown. You should follow the execution of the ffmpeg interop code and see where it crashes (it will probably be a FFMPEG call returning a negative value). It will probably tell you what is wrong. 

We haven't really given much attention towards the streaming part of this library yet. We were focusing on finishing "local" playback support first. You probably need to provide some config options to the rtmp stream. 

In any case, you can take a look at how VLC does it, a ffmpeg based solution should be pretty similar.

**Comment by lukasf** [comment]
From what I found out, ffmpeg does not support dash playback. Dash is not a video stream, it is a complex protocol which will allow you to open different versions of a stream and switch during playback. The stream itself could probably be decoded by ffmpeg, but ffmpeg will not do the dash protocol handling. So you cannot play dash using FFmpegInterop. If you cannot get this to play using normal MediaSource or AdaptiveMediaSource, you are out of luck. Closing this, since it is not a lib problem.

https://docs.microsoft.com/en-us/windows/uwp/audio-video-camera/adaptive-streaming
